scripts/run-ethtest-by-fork.py

```diff
@@ -51,57 +51,39 @@ def runtest_fork(input_directory, output_directory, fork='Shanghai', runtest_bin
-                    transaction_gaslimit = data[rootname]['transaction']['gasLimit']
-                    transaction_value = data[rootname]['transaction']['value']
-                    transaction = data[rootname]['transaction']
-
-                    test_index = 0
-                    for test in post_data[fork]:
-                        data = copy.deepcopy(data)
-                        data[rootname]['post'] = {fork: [test]}
-                        indexes = test.get('indexes', {})
-                        test_data_index = indexes.get('data')
-                        test_value_index = indexes.get('value')
-                        test_gas_index =  indexes.get('gas')
-
-                        new_transaction = copy.deepcopy(transaction)
-
-                        if test_data_index is not None:
-                            new_transaction['data'] = [ transaction_data[test_data_index], ]
-                            indexes['data'] = 0
-
-                        if test_value_index is not None:
-                            new_transaction['value'] = [ transaction_value[test_value_index], ]
-                            indexes['value'] = 0
-
-                        if test_gas_index is not None:
-                            new_transaction['gasLimit'] = [ transaction_gaslimit[test_gas_index], ]
-                            indexes['gas'] = 0
-
-                        output_filepath = os.path.join(output_directory, rel_path, f'{test_index}', filename)
-                        os.makedirs(os.path.dirname(output_filepath), exist_ok=True)
-
-                        data[rootname]['transaction'] = new_transaction
-
-                        with open(output_filepath, 'w', encoding='utf-8') as file:
-                            json.dump(data, file, ensure_ascii=False, indent=2)
-
-                        print(f"Processed and saved {output_filepath} successfully.")
-                        run_single_test(output_filepath, runtest_bin, geth_bin, cuevm_bin)
-                        test_index += 1
-                        if result:
-                            result['n_success'] += 1
-                            result['n_total'] += 1
+                    for rootname in list(data.keys()):
+                        transaction_data = data[rootname]['transaction']['data']
+                        transaction_gaslimit = data[rootname]['transaction']['gasLimit']
+                        transaction_value = data[rootname]['transaction']['value']
+                        transaction = data[rootname]['transaction']
+
+                        data_len = len(transaction_data)
+                        gaslimit_len = len(transaction_gaslimit)
+                        value_len = len(transaction_value)
+
+                        for data_index in range(data_len):
+                            for gas_index in range(gaslimit_len):
+                                for value_index in range(value_len):
```

Comment:
is this testing all the combinations instead of reading from `indexes`? Wouldn't this cause issue if we later decided to compare with the stateroot?

Reply:
We need to look at how the tests are done, but I think the post data is generated from the transaction data. So, all of the combinations will be part of the post data. Also, in our case, we are running to see the differences with the traces of other EVM implementations, so if there are more tests to verify, it is better. Also, @minhhn2910 said there are some tests without any post data.

Reply:
i think to avoid generating invalid tests, the best approach might be for tests with `post.[fork].indexes` use it to generate the tests, otherwise use the combinations of the `data`, `gaslimit`, `value` to create tests.